### PR TITLE
BUG: Behavior with fallback between raise and coerce #46071

### LINKED ISSUE #46071: BUG: to_datetime does not raise when errors='raise'.
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

s = pd.Series(['6/30/2025','1 27 2024'])
pd.to_datetime(s,errors='raise',infer_datetime_format=True)
```

### Issue Description

I'm confused about the interaction between `errors = {raise, coerce}` and `infer_datetime_format = {True, False}`.

1. From experiment 3, it tells me if I want to follow the format in first row using infer=True, the second row has a problem which causes it to be coerced to NaT. 

2. I also understand that in experiment 4, because infer=False, each row is free to have it's own format not necessarily following the format inferred from first row, so it doesn't have any problem being converted, thus not NaT.

3. **I don't understand why experiment 1 does not raise any error but converts row 2 properly?** 
(From experiment 3 i expect row 2 to be problematic when infer=True)

4. I understand experiment 2 converts with no problem because infer=False allows every row to have it's own format, similar reasoning as for experiment 4.

I tried 4 experiments on the same series of `s = pd.Series(['6/30/2025','1 27 2024'])`

**Experiment 1. errors = 'raise', infer_datetime_format=True**

**Input:** `pd.to_datetime(s,errors='raise',infer_datetime_format=True)`
**Output:**
```
0   2025-06-30
1   2024-01-27
dtype: datetime64[ns]
```

**Experiment 2. errors = 'raise', infer_datetime_format=False**

**Input:** `pd.to_datetime(s,errors='raise',infer_datetime_format=False)`
**Output: Same as experiment 1**

**Experiment 3. errors = 'coerce', infer_datetime_format=True** 

**Input:** `pd.to_datetime(s,errors='coerce',infer_datetime_format=True)`
**Output:**

```
0   2025-06-30
1          NaT
dtype: datetime64[ns]
```
**Experiment 4. errors = 'coerce', infer_datetime_format=False**

**Input:** `pd.to_datetime(s,errors='coerce',infer_datetime_format=False)`
**Output: Same as experiment 1**

### Expected Behavior

I expect the same error that caused experiment 3 to produce NaT in 2nd row with `'1 27 2024'`, to generate some error in experiment 1 for 2nd row too. 
I'm not sure what caused the NaT in experiment 3, but reasoning from the assumption that if `errors='coerce'` has found some problem, `errors='raise'` should find that problem too.
I also assume the NaT in experiment 3 is caused by infer_datetime_format by comparing experiment 3 and 4.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : bb1f651536508cdfef8550f93ace7849b00046ee
python           : 3.8.12.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.4.0
Version          : Darwin Kernel Version 20.4.0: Fri Mar  5 01:14:14 PST 2021; root:xnu-7195.101.1~3/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.1.1
setuptools       : 56.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.6.4
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.30.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.5.0
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.3
sqlalchemy       : 1.4.29
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : 2.0.1
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by srotondo**
- [ ] closes #46071 (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The problem reported by this issue is that success in parsing a date string depends on how errors are reported (coerce vs raise). How the error is reported should not affect whether or not a parsing error occurs. Initially it appears that experiment 1 should return an error (as experiment 3 does) because the second date string does not match the format established by the first string.  However, there is a fallback mechanism that is supposed to try parsing again without the effect of the `infer_datetime_format` flag. So in fact, experiment 3 should succeed in parsing the date as experiment 1 does.
 
The problem is that the fallback mechanism only works when `errors=raise`. This bug fix applies the same logic when `errors=coerce` so that the fallback parsing will take place regardless of how errors are reported.

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks for the PR @srotondo.

Do you mind reverting the unrelated changes (removing or adding blank lines) please? Also, we need to add a test that captures the problem, fails with the original code, and passes after your changes. The issue number, better have it as a comment in the test, not in the code.

Also, we need a release note with the change. You can have a look at any merged bug PRs as a reference.

**Comment by srotondo** [comment]
@datapythonista I've added the test and the release note to this code, is there anything else you would like me to do?

File: pandas/tests/arrays/test_datetimes.py [hunk 1]
```diff
@@ -639,3 +639,21 @@ def test_tz_localize_t2d(self):
 
         roundtrip = expected.tz_localize("US/Pacific")
         tm.assert_datetime_array_equal(roundtrip, dta)
+
+    @pytest.mark.parametrize(
+        "error",
+        ["coerce", "raise"],
+    )
+    def test_coerce_fallback(self, error):
+        # GH#46071
```
Comment (datapythonista) [inline pandas/tests/arrays/test_datetimes.py:648]: Thanks for adding this @srotondo. I don't think we're yet there, this doesn't seem to make so much sense.

My understanding is that with `infer_datetime_format=True` the format is inferred with the first non-NaN value, in this case `6/30/2025`, do the format should be `m/d/yyyy`. Then, when the next value is being parsed, it won't be able to, since `1 27 2024` is not a valid date in that format. When `errors='coerce'`, the returned Series should contain the first date, and the second `NaT`. When `errors='raise'` the `to_datetime` call should raise an exception.

This is very different to what you're asserting in this test, that it's that in both cases the result will be that both dates are being parsed.

Also, can you check if there is already a test for `to_datetime`? It should, it won't be testing this particular case, but maybe the same test can be reused in a way. Or at least have both tests next to each other.

**Review by datapythonista — CHANGES_REQUESTED** [review]
Sorry for the delay, and thanks for the changes @srotondo. Can you have a look at the comment please?

File: pandas/tests/arrays/test_datetimes.py (hunk 1 above)
Comment (srotondo) [reply pandas/tests/arrays/test_datetimes.py:648]: The original poster of this bug also thought that was how `infer_datetime_format` should work, but examining the programming closely makes it clear that it wants to avoid error returns as much as possible.

`infer_datetime_format` is meant to infer the first `datetime`'s format, and then cache that.  It will then attempt to use that inferred format on the subsequent `datetime`s to save time throughout the series.  If this format was made to be absolute for every term in the series, you would be correct that it should return the appropriate error result.  However, `infer_datetime_format` functions more like a guideline for the format.  There is then specific code in pandas that will fallback and try again to avoid errors for otherwise valid datetimes.  Raising an error or returning `NaT` should only happen if that fallback then proceeds to fail, for instance, if a datetime were out of range.

In terms of the test location, I could move this test to the `test_to_datetime.py` file, however, given that this test is mainly focused on the fallback itself and the piece of code that I changed wasn't directly in one of the `to_datetime` functions, but rather something it uses, I thought it would be more appropriate in the place I put it.  However, I'm completely open to moving it there.

Comment (datapythonista) [reply pandas/tests/arrays/test_datetimes.py:648]: I guess that's the behavior we want to keep then. But we need to fix the documentation of `to_datetime`, to make clear what `infer_datetime_format` does. And we need to test properly anyway. If it only raises with a date out of range or that doesn't make sense, then we need to add a case for that to the test.

**Comment by scott-r** [comment]
The fallback behavior is apparent in the code but not mentioned in the documentation for `infer_datetime_format`.

> infer_datetime_format: bool, default False
> If True and no format is given, attempt to infer the format of the datetime strings based on the first non-NaN element, and if it can be inferred, switch to a faster method of parsing them. In some cases this can increase the parsing speed by ~5-10x.

I think it would be accurate to add one more sentence at the end:

> If subsequent datetime strings do not follow the inferred format, parsing will fall back to the slower method of determining the format for each string individually.

That should be sufficient to remove the expectation that an error will be raised without committing to an exact description of the fallback parsing.

**Comment by srotondo** [comment]
@datapythonista I believe I have made the appropriate changes.  Is there anything else?

File: pandas/tests/arrays/test_datetimes.py [hunk 2]
```diff
@@ -639,3 +641,40 @@ def test_tz_localize_t2d(self):
...
+            assert result[1] is not pd.NaT
+
+        tm.assert_series_equal(expected, result)
+
+        expected2 = pd.Series([pd.Timestamp("2000-01-01 00:00:00"), pd.NaT])
+
+        es1 = pd.Series(["1/1/2000", "7/12/1200"])
+        es2 = pd.Series(["1/1/2000", "Hello"])
+
+        if error == "coerce":
```
Comment (datapythonista) [inline pandas/tests/arrays/test_datetimes.py:668]: Thanks for the fixes @srotondo. This test is a bit complex and not very easy to read. I think we could make things more readable if we have if for example split this in different tests. Something like:
- `test_to_datetime_infer_datetime_format_with_different_formats`
- `test_to_datetime_infer_datetime_format_with_nat`
- `test_to_datetime_infer_datetime_format_with_out_of_bounds`
- `test_to_datetime_infer_datetime_format_with_invalid_date`

What do you think? Then each test would be quite easy to read. Or maybe by parametrizing the input and the two outputs (coerce and raise). This will avoid a bit of unrepeated code and be quite concise, but probably not so readable.

See what makes sense to you, but would be good to make things as simple as possible, as the pandas codebase is already huge and too complex.

File: pandas/tests/arrays/test_datetimes.py [hunk 3]
```diff
@@ -639,3 +641,40 @@ def test_tz_localize_t2d(self):
...
+            assert result[1] is not pd.NaT
+
+        tm.assert_series_equal(expected, result)
+
+        expected2 = pd.Series([pd.Timestamp("2000-01-01 00:00:00"), pd.NaT])
+
+        es1 = pd.Series(["1/1/2000", "7/12/1200"])
+        es2 = pd.Series(["1/1/2000", "Hello"])
```
Comment (datapythonista) [inline pandas/tests/arrays/test_datetimes.py:666]: Maybe instead of `Hello` use `Invalid date` or something like that, so it's easier to understand what we're doing here.

**Review by datapythonista — CHANGES_REQUESTED** [review]
Thanks @srotondo, this makes more sense. I suggest some changes to make code more readable, see what you can do, and I'll review again when it's easier to see every case being tested.

File: pandas/tests/arrays/test_datetimes.py (hunk 2 above)
Comment (srotondo) [reply pandas/tests/arrays/test_datetimes.py:667]: @datapythonista thank you for your feedback on the tests.  Because the cases are all closely related, I think it will be easier to understand as one test, but I did simplify some of the test and added more comments to clarify parts of the test.

**Comment by srotondo** [comment]
@datapythonista I believe that I have made all the changes you have requested, including simplifying the tests.  Is there anything else you would like me to do?

**Review by datapythonista — CHANGES_REQUESTED** [review]
Sorry, I continue to find this test too hard to read. And very hard to understand what's wrong when it fails. I need to insist on using good testing practices, and restructure in a way that make tests atomic and simple. Like not having `if` statements, not having more than one expected value (unless using parametrization)...

If you don't find a better, way, using the previously suggested tests is a good start.

**Comment by srotondo** [comment]
@datapythonista I've split up the test into multiple pieces like you asked.

**Comment by datapythonista** [comment]
Thanks @srotondo, this is more readable now.

But looks like you're somehow reimplementing the tests in [here](https://github.com/pandas-dev/pandas/blob/main/pandas/tests/tools/test_to_datetime.py#L949), no? Would be good to start by adding the exact case or test that your fix addresses, which is not immediately clear to me. Thanks!

**Comment by srotondo** [comment]
> But looks like you're somehow reimplementing the tests in [here](https://github.com/pandas-dev/pandas/blob/main/pandas/tests/tools/test_to_datetime.py#L949), no? Would be good to start by adding the exact case or test that your fix addresses, which is not immediately clear to me. Thanks!

It might look like some of these tests are reimplementations, but there are some key differences in the cases that are being tested.  Given the line number you linked in that test, I assume you are mainly referring to these 3 tests in that file: `test_datetime_invalid_scalar`, `test_datetime_outofbounds_scalar`, and `test_datetime_invalid_index`.  While they do share a few similarities, there are some important key differences between these tests and my added tests.

To start with, `test_datetime_invalid_scalar` and `test_datetime_outofbounds_scalar` only deal in scalar values, unlike my tests, which deal with lists.  The difference from `test_datetime_invalid_index` lies largely in the data being tested.  The list used in `test_datetime_invalid_index` is entirely made up of values that will not parse, while the lists in my tests always start with a valid date.  This is important because `infer_datetime_format` only becomes relevant on the value after the first valid date, since it would need a format already to infer from.  Therefore this test doesn't really check the fallback code as my tests are meant to.

**Comment by datapythonista** [comment]
Thanks @srotondo that makes sense. Is there a reason why the tests you added are not next to the ones I mentioned? Seems like they should be next to each other. Or, not sure if it could make sense to add an extra assert in those lines to check the same cases when the invalid values are in a Series following a valid value.

And it's still not clear to me what's the exact case that you're fixing. Or are all the tests that you added failing without the change in the code function?

**Comment by srotondo** [comment]
@datapythonista The true failing case is that when 2 valid dates with different formats were passed in using the `errors = "coerce"` option and `infer_datetime_format = True`, `to_datetime` would return `NaT` for the second value, when it should correctly parse the second value.  The second and third tests I wrote ensure that `coerce` still returns `NaT` when it should return `NaT` and that the cases where `errors = "raise"` still raises exceptions when it should, respectively.

The choice of file for this test is mostly arbitrary, and I can move it if you'd like.  However, could you please tell me if there is anything else you would like me to change as well so that I can do all of it at the same time and we can get this bugfix merged.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
> I think it would be accurate to add one more sentence at the end:
> >    If subsequent datetime strings do not follow the inferred format, parsing will fall back to the slower method of determining the format for each string individually.

Hey - first, sorry to block this, and thanks for your work here

Just wanted to ask whether this is really what would be best - why not make `infer_datetime_format` strict:
- if the format is inferred, and some row doesn't conform to it, it just errors or coerces, rather than falling back to the slower path

This would involve removing the special-casing here

https://github.com/pandas-dev/pandas/blob/ac648eeaf5c27ab957e8cd284eb7e49a45232f00/pandas/core/tools/datetimes.py#L488-L499

and just have
```python
        if errors == "raise":
            raise
        elif errors == "coerce":
            result = np.empty(arg.shape, dtype="M8[ns]")
            iresult = result.view("i8")
            iresult.fill(iNaT)
        else:
            result = arg
```

This would also solve the linked issue, and rather than adding a new if-then statement, it'd remove one

**Comment by srotondo** [comment]
> Just wanted to ask whether this is really what would be best - why not make `infer_datetime_format` strict:
> 
> * if the format is inferred, and some row doesn't conform to it, it just errors or coerces, rather than falling back to the slower path

@MarcoGorelli There exists large amounts of code with the express purpose of causing the operation to fall back if the formats aren't the same.  The original documentation seems to strongly suggest that `infer_datetime_format` exists to cause parsing to speed up whenever possible, rather than causing otherwise valid dates to fail to parse.  Also, given how many different ways a date string can be written, forcing all inputs to `to_datetime` to be the same format doesn't seem like the best course of action to me.  I think that making this operation strict would go against the intended functionality.

**Comment by MarcoGorelli** [comment]
Currently, the docs say

>  if it can be inferred, switch to a faster method of parsing them

and don't specify what happens if the format can be inferred, but subsequent rows don't match that format - they don't say anything about the fallback behaviour.

And users don't expect the format to change midway, especially when using this option, e.g. https://github.com/pandas-dev/pandas/issues/46210

> You would expect no warning at all, since "infer_datetime_format=True" is already provided as an argument.

cc @mroeschke @jreback @jorisvandenbossche if you have comments

*TL;DR*: I'm suggesting that when using `infer_datetime_format=True`, the format detected from the first non-NaN value should be used to parse the rest of the Series, just like if the user had passed it to `format=`

*Example*
Concretely, I think the following two should behave the same:
```
pd.to_datetime(['01-31-2000', '20-01-2000'], infer_datetime_format=True, errors='raise')
pd.to_datetime(['01-31-2000', '20-01-2000'], format='%m-%d-%Y', errors='raise')
```
Currently, the former swaps format whilst the latter raises

**Comment by srotondo** [comment]
@MarcoGorelli Even though it's not very well documented, looking at the code and following the pathway makes it very clear that the fallback is intentional.  If you would like to change the behavior, I think that should be a separate PR.  This code change deals with the fact that `raise` and `coerce` were behaving inconsistently with `infer_datetime_format`, which has been fixed.  I think that removing the fallback behavior altogether is a much larger and separate issue.

**Review by MarcoGorelli — COMMENTED** [review]
Hey - just had a closer look, and indeed [the user guide](https://pandas.pydata.org/docs/user_guide/io.html#inferring-datetime-format) does explicitly note:

> pandas will fallback to the usual parsing if either the format cannot be guessed or the format that was guessed cannot properly parse the entire column of strings

That's fine then, I'll open a separate issue about changing that

File: pandas/tests/arrays/test_datetimes.py [hunk 4]
```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
...
+
+    def test_fallback_with_errors_raise(self):
+        # GH#46071
+        # Invalid inputs
+        # Parsing should fail for the second element
+        dates1 = pd.Series(["1/1/2000", "7/12/1200"])
+        with pytest.raises(
+            OutOfBoundsDatetime, match="Out of bounds nanosecond timestamp"
+        ):
+            pd.to_datetime(dates1, errors="raise", infer_datetime_format=True)
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/test_datetimes.py:683]: Isn't this duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L961-L987

?

File: pandas/tests/arrays/test_datetimes.py [hunk 5]
```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
...
+        expected = pd.Series(
+            [pd.Timestamp("2025-06-30 00:00:00"), pd.Timestamp("2024-01-27 00:00:00")]
+        )
+        result = pd.to_datetime(s, errors=error, infer_datetime_format=True)
+        tm.assert_series_equal(expected, result)
+
+    @pytest.mark.parametrize(
+        "dateseries",
+        [
+            pd.Series(["1/1/2000", "7/12/1200"]),
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/test_datetimes.py:663]: isn't this one duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L961-L987

?

File: pandas/tests/arrays/test_datetimes.py [hunk 6]
```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
...
+        # GH#46071
+        # Invalid inputs
+        # Parsing should fail for the second element
+        dates1 = pd.Series(["1/1/2000", "7/12/1200"])
+        with pytest.raises(
+            OutOfBoundsDatetime, match="Out of bounds nanosecond timestamp"
+        ):
+            pd.to_datetime(dates1, errors="raise", infer_datetime_format=True)
+
+        dates2 = pd.Series(["1/1/2000", "Invalid input"])
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/test_datetimes.py:685]: isn't this duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L989-L1012

?

File: pandas/tests/arrays/test_datetimes.py [hunk 7]
```diff
@@ -639,3 +641,47 @@ def test_tz_localize_t2d(self):
...
+            [pd.Timestamp("2025-06-30 00:00:00"), pd.Timestamp("2024-01-27 00:00:00")]
+        )
+        result = pd.to_datetime(s, errors=error, infer_datetime_format=True)
+        tm.assert_series_equal(expected, result)
+
+    @pytest.mark.parametrize(
+        "dateseries",
+        [
+            pd.Series(["1/1/2000", "7/12/1200"]),
+            pd.Series(["1/1/2000", "Invalid input"]),
```
Comment (MarcoGorelli) [inline pandas/tests/arrays/test_datetimes.py:664]: isn't this duplicative of

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L989-L1012

?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
The fix itself looks good to me, thanks! I just have some comments about the tests:

1. is the location right? Wouldn't `/pandas/tests/tools/test_to_datetime.py` be better?
2. couldn't we just cover all these new tests by parametrising over `errors` in  `/pandas/tests/tools/test_to_datetime.py::TestToDatetimeInferFormat::test_to_datetime_infer_datetime_format_inconsistent_format` ?

Apologies for how long reviews are taking, I'll do my best to help push this across the line now

File: pandas/tests/arrays/test_datetimes.py (hunk 7 above)
Comment (srotondo) [reply pandas/tests/arrays/test_datetimes.py:664]: Ok, that's a lot of comments there, so I'm going to try to answer all these questions here in this response here.  Please tell me if it's not clear which comment I'm talking about at any given point here.

I was asked about these last 4 comments before, so I'll link my previous response here so you can see it in case you missed it.
> It might look like some of these tests are reimplementations, but there are some key differences in the cases that are being tested. Given the line number you linked in that test, I assume you are mainly referring to these 3 tests in that file: `test_datetime_invalid_scalar`, `test_datetime_outofbounds_scalar`, and `test_datetime_invalid_index`. While they do share a few similarities, there are some important key differences between these tests and my added tests.
> 
> To start with, `test_datetime_invalid_scalar` and `test_datetime_outofbounds_scalar` only deal in scalar values, unlike my tests, which deal with lists. The difference from `test_datetime_invalid_index` lies largely in the data being tested. The list used in `test_datetime_invalid_index` is entirely made up of values that will not parse, while the lists in my tests always start with a valid date. This is important because `infer_datetime_format` only becomes relevant on the value after the first valid date, since it would need a format already to infer from. Therefore this test doesn't really check the fallback code as my tests are meant to.

In terms of your comments about the code changes and the first test comment, I agree with those changes and I can do that for sure.  Also, in terms of test location, I'm fully open to changing that, since the choice I made in location was largely arbitrary.

Comment (MarcoGorelli) [reply pandas/tests/arrays/test_datetimes.py:664]: Thanks for explaining and for your attention to detail, that's very clear! Apologies for having missed that, I'd looked through in-line comments but had missed this one in the main thread

For test location - does `pandas/tests/arrays/test_datetimes.py::TestToDatetimeInferFormat` look good to you?

Comment (MarcoGorelli) [reply pandas/tests/arrays/test_datetimes.py:664]: @srotondo have you had a look at https://github.com/pandas-dev/pandas/pull/48621? 

First, because you've had a close look at this part of the code so I'd really value your input

Second, because if we did go ahead with that, then it'd make the word in this PR unnecessary

Comment (srotondo) [reply pandas/tests/arrays/test_datetimes.py:664]: > For test location - does `pandas/tests/arrays/test_datetimes.py::TestToDatetimeInferFormat` look good to you?

To clarify, is this a new section in `test_datetimes.py` that you are suggesting, because as far as I can tell, this doesn't actually exist yet.

Comment (MarcoGorelli) [reply pandas/tests/arrays/test_datetimes.py:664]: I meant to add new tests to the class which

https://github.com/pandas-dev/pandas/blob/a712c5019dc0cfb58652ddcdf06361244f38ad9f/pandas/tests/tools/test_to_datetime.py#L989-L1012

belongs to

**Comment by MarcoGorelli** [comment]
PDEP4 was accepted, and it looks like the implementation is almost in

Let's close this then

Thanks @srotondo for your PR - you've done great work, and I'm sorry to be closing it, I just think a different direction is a better direction for the future of pandas
